pkg/gofr/container/datasources.go

```diff
@@ -174,3 +180,7 @@ type provider interface {
 	// Connect establishes a connection to Cassandra and registers metrics using the provided configuration when the client was Created.
 	Connect()
 }
+
+type HealthChecker interface {
+	HealthCheck() interface{}
```

Comment:
This interface should pass context.


```suggestion
	HealthCheck(context.Context) any
```

Especially if you consider your heal checker has the context.

Some resources might need a context, or should have one. It would be cleaner

Reply:
This would make more sense, thanks